src/orders/presentation/orders.controller.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpCode,
+  HttpException,
+  HttpStatus,
+  Inject,
+  Param,
+  Post,
+  Put,
+} from "@nestjs/common";
+import { IOrderService } from "../application/orders.service.interface";
+import { Order } from "../domain/order.entity";
+import { CreateOrUpdateOrdersDto } from "../dto/createOrUpdateOrders.dto";
+import { OrderAlreadyExistsException } from "../exception/OrdersAlreadyExistsException.exception";
+
+@Controller("orders")
+export class OrdersController {
+  constructor(@Inject("IOrderService") private readonly orderService: IOrderService) {}
+
+  @Post()
+  @HttpCode(200)
+  async create(@Body() createOrUpdateOrdersDto: CreateOrUpdateOrdersDto): Promise<Order> {
+    try {
+      return await this.orderService.create(createOrUpdateOrdersDto as Order);
+    } catch (error) {
+      if (error instanceof OrderAlreadyExistsException) {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+      throw new HttpException("Internal Server Error", HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  @Get()
+  @HttpCode(200)
+  async getAllOrder(): Promise<Order[]> {
+    try {
+      return await this.orderService.findAll();
+    } catch (error) {
+      throw new HttpException("Internal Server Error", HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
Peut etre ajouter un check de l'error pour differencier un not found d'une internal error